SystemReference/Payload/Camera/Camera.cpp

```diff
@@ -6,117 +6,304 @@
+    CameraComponentBase::init(queueDepth, instance);
+    camManager = std::make_unique<libcamera::CameraManager>();
   }
-  m_capture.open(deviceIndex);
-  if (!m_capture.isOpened()) {
+  // start camera manager and acquire camera
+  // return true on success, false otherwise
+  bool Camera::open(I32 deviceIndex) {
+    int returnCode = camManager->start();
+    // check to see if the camera manager was started successfully
+    if (returnCode == 0) {
+      // if no cameras were detected, emit a CameraNotDetect event
+      if(camManager->cameras().empty()) {
+        this->log_WARNING_HI_CameraNotDetected();
+        return false;
+      }
+      // get the camera from the CameraManager and acquire it
+      m_capture = camManager->cameras()[deviceIndex];
+      returnCode = m_capture->acquire();
+
+      // camera was acquired successfully
+      if (returnCode == 0) {
+        return true;
+      }
+      // camera was previously acquired, emit a CameraAlreadyOpen event
+      else if (returnCode == EBUSY) {
+        this->log_ACTIVITY_LO_CameraAlreadyOpen();
+        return true;
+      }
+    }
+    // some other error occurred, emit a CameraOpenError event
     this->log_WARNING_HI_CameraOpenError();
     return false;
   }
-#endif
-  return true;
-}
-
-Camera ::~Camera() {}
-
-// ----------------------------------------------------------------------
-// Command handler implementations
-// ----------------------------------------------------------------------
-
-void Camera ::TakeAction_cmdHandler(const FwOpcodeType opCode, const U32 cmdSeq,
-                                    Payload::CameraAction cameraAction) {
-  RawImageData rawImageData;
-  Fw::Buffer imgBuffer;
-#ifdef USES_OPENCV
-  std::vector<uchar> buffer;
-  cv::Mat frame;
-  m_capture.read(frame);
-
-  if (frame.empty()) {
-    this->log_WARNING_HI_BlankFrame();
-    return;
+
+  Camera ::~Camera() {}
+
+  void Camera ::parameterUpdated(FwPrmIdType id) {
+    // Read the parameter value
+    Fw::ParamValid isValid;
+    ImgResolution resolution = this->paramGet_IMG_RESOLUTION(isValid);
+    FW_ASSERT(isValid == Fw::ParamValid::VALID, isValid);
+
+    if(PARAMID_IMG_RESOLUTION == id) {
+      // update the camera configuration to use the updated IMG_RESOLUTION parameter value
+      setCameraConfiguration(resolution);
+    }
   }
 
-  // Set up buffer for image data
-  U32 imgSize = frame.rows * frame.cols * frame.elemSize();
-  imgBuffer = allocate_out(0, imgSize);
-  if (imgBuffer.getSize() < imgSize){
-    this->log_WARNING_HI_InvalidBufferSizeError(imgBuffer.getSize(), imgSize);
-    this->deallocate_out(0, imgBuffer);
-    return;
+  libcamera::Request *requestReceived;
+  void Camera ::requestComplete(libcamera::Request *request) {
+    if (request->status() == libcamera::Request::RequestCancelled) {
+      return;
+    }
+    requestReceived = request;
+  }
+
+  void Camera ::allocateBuffers() {
+    printf("Allocating buffers\n");
```

Comment:
Printf should be removed or made Os::Log

Reply:
Removed prints